src/couch_replicator/src/couch_replicator_docs.erl

```diff
@@ -271,15 +275,26 @@ before_doc_update_owner(Other, Name, Db, #doc{body = {Body}} = Doc) ->
 after_doc_read(#doc{id = <<?DESIGN_DOC_PREFIX, _/binary>>} = Doc, _Db) ->
     Doc;
 after_doc_read(#doc{body = {Body}} = Doc, Db) ->
-    #user_ctx{name = Name} = couch_db:get_user_ctx(Db),
+    Ctx = #user_ctx{roles = Roles, name = Name} = couch_db:get_user_ctx(Db),
+    IsReplicator = lists:member(?REPLICATOR_ROLE, Roles),
+    % Internal replicator opens dbs with the ?ADMIN_CTX macro
+    IsInternalRepl = (Ctx == ?ADMIN_USER),
     case (catch couch_db:check_is_admin(Db)) of
         ok ->
             Doc;
         _ ->
             case get_value(?OWNER, Body) of
                 Name ->
                     Doc;
+                _Other when (IsReplicator orelse IsInternalRepl) ->
+                    % When the replicator itself updates the doc don't patch it with a fake
+                    % rev and strip the creds
+                    Doc;
                 _Other ->
+                    % Not admin, user name doesn't match, and it's not the
+                    % replicator app: this is another user reading it. Strip
```

Comment:
Since this is one user reading a different user's replication document, it's not clear to me why we present a mangled view of the document rather than say return a 404 or 403. I know we discussed this privately, and you were also not sure, but I wanted to raise this issue here in public so others who may know the answer would be able to answer the question. It would be great if the comment could be extended to explain the reasoning, assuming we still want to keep this peculiar behavior.

Reply:
I think at this point it's sort of legacy behavior that we're trying to preserve for compatibility. But yeah, it would be interesting to find out why we opted to redacting vs a 404 though